### PR TITLE
[#1699] improvement: Upgrade from commons-collections:commons-collections:3.2.2 to org.apache.commons:commons-collections:4.4

### PR DESCRIPTION
**Pull request by rickyma**
### What changes were proposed in this pull request?

Upgrade from commons-collections:commons-collections:3.2.2 to org.apache.commons:commons-collections:4.4.

### Why are the changes needed?

Fix: https://github.com/apache/incubator-uniffle/issues/1699.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing UTs.

### CODE REVIEW

**Comment by rickyma** [comment]
PTAL. @jerqi @zuston

**Comment by zuston** [comment]
Is this common update for other opensource projects?

**Comment by rickyma** [comment]
I think it is. You can refer to https://issues.apache.org/jira/browse/SPARK-37968.

**Comment by jerqi** [comment]
@LuciferYang  Could you help me review this pull request?

File: pom.xml [hunk 1]
```diff
@@ -56,7 +56,7 @@
     <version.checksum-maven-plugin>1.11</version.checksum-maven-plugin>
     <awaitility.version>4.2.0</awaitility.version>
     <checkstyle.version>9.3</checkstyle.version>
-    <commons-collections.version>3.2.2</commons-collections.version>
+    <commons-collections4.version>4.5.0-M1</commons-collections4.version>
```
Comment (LuciferYang) [inline pom.xml:59]: I suggest migrating to 4.4 first, rather than a milestone version

Comment (rickyma) [reply pom.xml]: Done.

File: server/pom.xml [hunk 2]
```diff
@@ -129,13 +129,12 @@
       </exclusions>
     </dependency>
     <dependency>
-      <groupId>commons-collections</groupId>
-      <artifactId>commons-collections</artifactId>
-      <version>${commons-collections.version}</version>
+      <groupId>org.apache.commons</groupId>
```
Comment (LuciferYang) [inline server/pom.xml:132]: should only modify the `commons-collections` part, no need to touch the `commons-lang3` part

Comment (LuciferYang) [reply server/pom.xml:132]: Others is fine to me. But please check whether there are cascading dependencies on commons-collections 3. If there are, whether it is necessary to retain the definition of commons-collections 3 version in uniffle. @rickyma

Comment (rickyma) [reply server/pom.xml:132]: I just changed the order of the dependencies. `commons-lang3` remains the same as before. 

The UTs have all passed, so the casading dependencies should be OK.

Comment (LuciferYang) [reply server/pom.xml:132]: OK

Comment (rickyma) [reply server/pom.xml:132]: We can find commons-collections 3 in the dependency tree which is introduced by `hadoop-common`, but it is provided:
```
[INFO] +- org.apache.hadoop:hadoop-common:jar:2.8.5:provided
[INFO] |  +- commons-collections:commons-collections:jar:3.2.2:provided
```
@LuciferYang

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by LuciferYang** [comment]
Merged into mater. Thanks @rickyma

**Pull request merged**